packages/bindings/src/types.rs

```diff
@@ -23,14 +23,14 @@ impl Swap {
 #[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
 pub struct Step {
     pub pool_id: u64,
-    pub denom_out: String,
+    pub denom: String,
```

Comment:
I don't like this change, it makes it more confusing.

A step has an in denom, and out denom and a pool. When we only use `denom`, it is unclear which one we define and which we inherit from the last step or next step

Reply:
Do you think it is better to:

* always use out?
* alternately use in and out?

If we make it change, then we should have two types for this.

I prefer to make the route construction the same (cleaner to the caller) and then transform this in the Go code to what the osmosis binary exposes. There is a deterministic transformation possible.

Basically, when wrapping their API, we can make some changes to improve it as well, as we are writing the wrapper code already.

Reply:
I agree on improving their API, but we must be careful with the meaning of the fields.

All right, I'll revert this. In the mean time, I've put a TODO / error for multi-hop support in estimating swap price, as you suggested.